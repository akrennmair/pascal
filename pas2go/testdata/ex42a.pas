program ex42a;

var i : integer;
	h, t, o : integer;

function fac(i : integer) : integer;
begin
	if i <= 1 then
		fac := 1
	else
		fac := i * fac(i - 1)
end;

begin
	for i:=100 to 999 do
	begin
		h := i div 100;
		t := (i mod 100) div 10;
		o := i mod 10;
		if i = (fac(h) + fac(t) + fac(o)) then
			writeln(i, ' = ', h, '! + ', t, '! + ', o, '!')
	end
end.
