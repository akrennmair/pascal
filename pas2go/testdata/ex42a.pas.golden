package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program ex42a
func main() {
	var (
		i int
		h int
		t int
		o int
	)
	_ = i
	_ = h
	_ = t
	_ = o

	var fac func(i int) int
	fac = func(i int) (fac_ int) {
		if i <= 1 {
			fac_ = 1
		} else {
			fac_ = i * fac(i-1)
		}
		return
	}

	for i = 100; i <= 999; i++ {
		h = i / 100
		t = (i % 100) / 10
		o = i % 10
		if i == fac(h)+fac(t)+fac(o) {
			system.Writeln(i, " = ", h, "! + ", t, "! + ", o, "!")
		}
	}
}
